piet/src/gradient.rs

```diff
@@ -1,25 +1,29 @@
 //! Gradient specifications.
 
-use kurbo::Vec2;
+use std::borrow::Cow;
+
+use kurbo::{Point, Rect, Vec2};
+
+use crate::{IBrush, RenderContext};
 
 use crate::Color;
 
 /// Specification of a gradient.
 #[derive(Clone)]
-pub enum Gradient {
+pub enum RawGradient {
```

Comment:
With the added complexity of the new 'GradientDescription' type, we want to make sure that the different between the description and the specific gradient is clear, both in naming and documentation.

I think that `Gradient` is the best name for the GradientDescription type, presuming it will be used far more frequently.

I'm less sure about the name for this type. I assume the reasoning here is that `RawGradient` is less abstract/lower level, but `Raw` also has the implication of being unprocessed, which is sort of the opposite of how we're using it.

What we're really saying here is that we have taken a gradient description and resolved it to a specific gradient, based on some concrete geometry. This gradient is (generally) no longer suitable to be reused for shapes other than the shape it was resolved for.

Names that come to mind that capture this more fully are `ResolvedGradient`, `CookedGradient`, `ConcreteGradient`, or `LiteralGradient`. Maybe `FixedGradient`.

The only major argument I could see for `RawGradient` is concision; if we expect this type to be used frequently elsewhere then it might be worth using a shorter name.

In either case, the documentation needs to clearly state the difference between the two forms.

Reply:
Of these, I like FixedGradient best, as it actually says what the difference is. I agree on the documentation and will address that.